File: services/api/src/store.ts
// Copyright 2026 Satoshi Maemoto
// SPDX-License-Identifier: Apache-2.0

import type { SessionState } from "@nakanaori/agents";

const sessions = new Map<string, SessionState>();

export function put(session: SessionState): void {
  sessions.set(session.session_id, session);
}

export function get(sessionId: string): SessionState | undefined {
  return sessions.get(sessionId);
}

export function listAll(): SessionState[] {
  return [...sessions.values()];
}

File: services/api/nakanaori_api/store.py
"""In-memory session store (MVP). Replace with Firestore for production."""

from nakanaori.orchestrator import SessionState

_sessions: dict[str, SessionState] = {}


def put(session: SessionState) -> None:
    _sessions[session.session_id] = session


def get(session_id: str) -> SessionState | None:
    return _sessions.get(session_id)
